foundation/foundation-mda/src/test/java/com/boozallen/aiops/mda/generator/PropertiesStep.java

```diff
@@ -0,0 +1,76 @@
+package com.boozallen.aiops.mda.generator;/*-
+ * #%L
+ * aiSSEMBLE::Foundation::MDA
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aiops.mda.metamodel.element.*;
+import io.cucumber.java.Before;
+import io.cucumber.java.Scenario;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.fermenter.mda.GenerateSourcesHelper;
+import org.technologybrewery.fermenter.mda.element.ExpandedProfile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class PropertiesStep extends AbstractModelInstanceSteps {
+    private static final Logger logger = LoggerFactory.getLogger(PropertiesStep.class);
+
+    @Before("@properties-generation")
+    public void setup(Scenario scenario) throws IOException {
+        this.scenario = scenario.getName();
+        FileUtils.deleteDirectory(GENERATED_METADATA_DIRECTORY);
```

Comment:
A: I would have expected this to be part of a teardown step not setup.  If this was necessary, it may mean we have some other test that isn't cleaning up after itself properly.

Reply:
yup you are correct, that was legacy of my copy pasta. Thank you for pointing out cleaning up..